commit-crimes/shelf-life
# Feat/implement overview screen

# Pull Request: Implement Overview Screen and Household Management Enhancements

## Branch: feat/implement-OverviewScreen

### Description:

This pull request introduces several key enhancements to the ShelfLife application, including the implementation of the Overview Screen, the addition of household management features, and improvements to the overall user experience. The changes involve new UI components, modifications to existing code, and integration of Firebase functionality for managing households.

### Key Changes:

1. Overview Screen Implementation:
	* Added a new OverviewScreen to provide a summary view of household and food items.
	* Integrated navigation support from the authentication screen to the Overview Screen upon successful sign-in.
	* Modified MainActivity to include the Overview Screen in the navigation graph.
2. Household Management:
	* Created a HouseHold data class representing a household, including its name, members, and associated food items.
	* Developed the HouseHoldRepositoryFirestore class for Firebase integration to manage household data.
	* Implemented CRUD (Create, Read, Update, Delete) operations for households using Firestore.
	* Added a HouseholdViewModel to manage household-related data and state within the application.

3. UI Enhancements:

	* Top Navigation Menu:
	* Introduced a TopNavigationMenu.kt to allow easy navigation between different sections of the app.
	* Added functionality for household switching via a sidebar.
	* Integrated a search bar for filtering food items within households.
	* Edit and Add Household Popups:
	* Added AddHouseHoldPopUp.kt to facilitate the creation of new households.
	* Added EditHouseHoldPopUp.kt to allow editing of household names and deletion of selected households.
4. Food Item Enhancements:
	* Updated the FoodItem data model to streamline the location attribute.
	* Refactored the FoodItemRepositoryFirestore to improve data handling for food items associated with households.
	* Implemented new methods in ListFoodItemsViewModel to handle updates to the food item list.
5. Design and Theming Updates:
	* Modified the app’s color scheme to utilize a green-themed palette for both light and dark modes, enhancing visual consistency.
	* Updated BottomNavigationMenu.kt to better align with the new design and provide a smoother user experience.
6. Bug Fixes:
	* Fixed issues in the menu drawer logic to ensure smooth navigation within the app.
	* Addressed bugs related to the food item location logic and household member list handling.
	* Improved error handling and logging in the Firebase repository classes to ensure clearer debugging and maintenance.

### Files Added/Modified:

New Files:
	* TopNavigationMenu.kt
	* AddHouseHoldPopUp.kt
	* EditHouseHoldPopUp.kt
	* HouseHold.kt, HouseHoldRepository.kt, HouseHoldRepositoryFirestore.kt
	* HouseholdViewModel.kt
	* OverviewScreen.kt
Modified Files:
	* MainActivity.kt: Updated navigation setup.
	* FoodItem.kt: Refactored data model for location handling.
	* FoodItemRepositoryFirestore.kt: Improved data conversion and handling.
	* ListFoodItemsViewModel.kt: Added new methods for food item management.
	* Theme.kt, Color.kt: Updated color schemes for a new design language.
	* SignIn.kt: Redirected post-sign-in navigation to the Overview Screen.

### How to Test:

1. Sign-In Flow: Start the app and sign in with your credentials. Ensure that the app navigates to the Overview Screen after successful sign-in with either the first time sign in or just the regular overview.
2. Overview Screen: Verify that the Overview Screen displays the correct household and food item information.
3. Household Management:
	* Test adding, editing, and deleting households using the respective popups.
	* Confirm that the household changes reflect accurately in the UI and Firebase.
4. Food Item Handling: Check that the food items are correctly displayed and categorised according to their storage location.
5. Navigation Testing: Use the sidebar and top navigation menu to switch between different sections of the app, ensuring smooth transitions.

### Additional Notes:
	* This pull request includes significant code refactoring to streamline household and food item handling logic.
	* Future improvements may include optimisations for loading times when fetching large lists of households and food items from Firebase.
	* In the future tests and test tags have to be added for most of the code

### Checklist:
	* Code is well-documented with comments and follows project conventions.
	* All new features have been tested for both functionality and UI design.
	* Firebase operations have been verified to ensure data consistency.
	* Existing unit tests have been updated, and new tests have been added where necessary.

### Reviewers:

Please review the changes and provide feedback or approval for merging into the main branch. Thank you!
